upload-api/html.js

```diff
@@ -0,0 +1,215 @@
+import { render } from 'preact-render-to-string'
+
+/**
+ * Build HTML document
+ *
+ * @param {string} body
+ */
+export function buildDocument(body) {
+  return `
+<!doctype html>
+<html class="no-js" lang="">
+
+<head>
+  <meta charset="utf-8">
+  <title>w3up Email Validation</title>
+  <meta name="description" content="">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@acab/reset.css">
+  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/water.css@2/out/dark.min.css">
+  <style>
+    :root {
+      --background-body: hsl(212deg 27% 17%);
+      --background: hsl(212deg 27% 12%);
+      --background-alt: hsl(212deg 27% 14%);
+      --button-base: hsl(212deg 27% 10%);
+      --button-hover: hsl(212deg 27% 6%);
+      --scrollbar-thumb: hsl(212deg 27% 10%);
+      --scrollbar-thumb-hover: hsl(212deg 27% 6%);
+    }
+    :where(:root) body {
+      display: grid;
+      height: 100vh;
+      height: 100dvh;
+      padding: 0;
+      margin: 0;
+      max-width: 100%;
+    }
+    body {
+      margin: 0 40px;
+      padding: 40px 0;
+    }
+    .fcenter {
+      display: flex;
+      align-items: center;
+      flex-direction: column;
+      justify-content: center;
+    }
+    .mcenter {
+      margin: 0 auto;
+    }
+    .box {
+      max-width: 640px;
+      background-color: var(--background-alt);
+      padding: 20px;
+      margin: 1em 0;
+      border-radius: 6px;
+      overflow: hidden;
+    }
+    .box > p {
+      margin-bottom: 10px;
+    }
+    .box > p:last-child {
+      margin-bottom: 0;
+    }
+  </style>
+</head>
+<body>
+${body}
+</body>
+</html>`
+}
+
+export class HtmlResponse extends Response {
+  /**
+   *
+   * @param {import('preact').VNode<{}>} body
+   * @param {ResponseInit} [init]
+   */
+  constructor(body, init = {}) {
+    const headers = {
+      headers: {
+        'content-type': 'text/html; charset=utf-8',
+      },
+    }
+    super(buildDocument(render(body)), { ...init, ...headers })
```

Comment:
yep agree - if we kept these here in the medium-term I think it might be worth moving in this direction, but this is hopefully a short-term bit of ugliness.